source/macos/AdaptiveCards/AdaptiveCards/Views/ACRDateField.swift

```diff
@@ -122,6 +138,13 @@ class ACRDateField: NSView, InputHandlingViewProtocol {
         addTrackingArea(trackingArea)
     }
     
+    @objc private func handleClearAction() {
+        textField.stringValue = ""
+        dateValue = ""
+        selectedDate = .none
+        clearButton.isHidden = true
+    }
```

Comment:
Why are we setting `isHidden` property in multiple places. It should have only one single source of truth.

```suggestion
        dateValue = nil
    }
```
This should be enough here.

Reply:
we need to also set the textfield.stringvalue and clearButton.isHidden here
This handles the general case and the other clearButton.isHidden = true is for the specific case when there is an initial value already stored